dev/tests/unit/Mage/Core/Helper/EnvironmentConfigLoaderTest.php

```diff
@@ -0,0 +1,153 @@
+<?php
+declare(strict_types=1);
+
+namespace OpenMage\Tests\Unit\Core\Helper;
+
+use PHPUnit\Framework\TestCase;
+use Mage_Core_Helper_EnvironmentConfigLoader;
+use Mage_Core_Model_Config;
+
+class EnvironmentConfigLoaderTest extends TestCase
+{
+    protected const ENV_CONFIG_DEFAULT_PATH = 'OPENMAGE_CONFIG__DEFAULT__GENERAL__STORE_INFORMATION__NAME';
+    protected const ENV_CONFIG_DEFAULT_VALUE = 'default_new_value';
+    protected const ENV_CONFIG_WEBSITE_PATH = 'OPENMAGE_CONFIG__WEBSITES__BASE__GENERAL__STORE_INFORMATION__NAME';
+    protected const ENV_CONFIG_WEBSITE_VALUE = 'website_new_value';
+    protected const ENV_CONFIG_STORE_PATH = 'OPENMAGE_CONFIG__STORES__GERMAN__GENERAL__STORE_INFORMATION__NAME';
+    protected const ENV_CONFIG_STORE_VALUE = 'store_german_new_value';
+
+    public function setup(): void
+    {
+        \Mage::setRoot('');
+    }
+
+    public function testBuildPath()
+    {
+        $environmentConfigLoaderHelper = new Mage_Core_Helper_EnvironmentConfigLoader();
+        $path = $environmentConfigLoaderHelper->buildPath('GENERAL', 'STORE_INFORMATION', 'NAME');
+        $this->assertEquals('general/store_information/name', $path);
+    }
+
+    public function testBuildNodePath()
+    {
+        $loader = new Mage_Core_Helper_EnvironmentConfigLoader();
+        $nodePath = $loader->buildNodePath('DEFAULT', 'general/store_information/name');
+        $this->assertEquals('default/general/store_information/name', $nodePath);
+    }
+
+    /**
+     * @dataProvider envOverridesDataProvider
+     *
+     */
+    public function testEnvOverrides(array $config)
+    {
+        error_reporting(0);
+        $xmlStruct = $this->getTestXml();
+
+        $xmlDefault = new Mage_Core_Model_Config($xmlStruct);
+        $xmlDefault = $xmlDefault->loadModulesConfiguration('config.xml', $xmlDefault);
+        $xml = new Mage_Core_Model_Config($xmlStruct);
+        $xml = $xml->loadModulesConfiguration('config.xml', $xml);
```

Comment:
To fix the error 

> TypeError: Cannot access offset of type string on string

I tested the following code snippet (**I do not really know what I am doing, just trial and error**):

```suggestion
        $xml = new Varien_Simplexml_Config();
        $xml->loadString($xmlStruct);
        $config = Mage::getConfig()->loadModulesConfiguration('config.xml', $xml);
        $r = $config->getNode('default/general/store_information/name');
```

$r is
```
 object(Varien_Simplexml_Element)#177 (1) {
  [0] => string(12) "test_default"
}
```

Hope it helps you to fix the errors.